package.json

```diff
@@ -43,7 +43,7 @@
     "web-vitals": "^2.1.0"
   },
   "scripts": {
-    "start": "GENERATE_SOURCEMAP=false react-scripts start",
```

Comment:
-	    "build": "cross-env GENERATE_SOURCEMAP=false react-scripts build" has been added

Reply:
Hey guys, didn't want to bombard you with comments, so here is everything. "-" indicates answer to above line. Also, we merge with latest changes on development, so the bug Emma fixed is taken care of.


maybe this solution: https://stackoverflow.com/a/61967190/1200070 ?
-	    "build": "cross-env GENERATE_SOURCEMAP=false react-scripts build" have been added

-           also removed .env file
 
remove this pls.
-	Removed Zeeguu_API.prototype.AUDIO_EXP_COMPLETED = "AUDIO_EXP_COMPLETED";

so the audio exercise message is open allways? this looks suspicious.
-	Added conditional:
if (!LocalStorage.displayedAudioExperimentPopup()) {
      setAudioExerciseMessageOpen(true);
    }

would be nice to have all of these start withe the same prefix?
-	Changed all key values to start with audio_*

displayedAudioExperimentPopup and setDisplayedAudioExperimentPopup are not really used in any meaningful way in this component. Why are they not state in the AudioExerciseMessage?
-	Moved to AudioExerciseMessage and simplified useeffect

temp is not used?
-	Removed

at this point in the code, it's very clear that you're referring to the audio experiment so you can consider the local variable having a shorter name numberOfSessions.
-	Updated var name
also, simplify the conditional
-	Simplified


finally, do what you say: if your function tells me that you "check", then don't modify the state!
-	Renamed function to checkAndUpdateAudioExperimentCompleted? 

is equivalent to simply return noofsessions
-	Updated to always return noofsessions

Ditto
-	Updated to always return noofsessions

what is the meaning of the highlighting colors? why are some props a very pale color?
-	They’re not being used, removed them from the code

maybe log the event also to the server? in case they close the popup without doing the survey you can remind them with an email later.
-	Have been implemented

this and the next don't change state ever in this component. why are they state? you can make them as state if you need them in QuestionnaireMessage.
-	Refactored, same with audiomessage.js
use logUserActivity instead of readerActivity. Reader is for the article "reader" part of the system.
-	Have been implemented

here you see the limitations of the check...Completed function name.
separate behavior in two functions: is...Completed and set...Completed. Then use them accordingly.
-	Renamed function name to: checkAndUpdateAudioExperimentCompleted?



dito.
-	Renamed function name to: checkAndUpdateAudioExperimentCompleted?

thank for fixing this!
-	 Np

it's hard for me to see exactly what happens here, but there seems to be A LOT of duplication.
if i'm right, the only difference between the two blocks is that if handleClick is defined, you call it and otherwise not? Couldn't you fix this by just adding the condition inside the onClick() from SpeakButton? e.g. something like:
-	I believe the reason we have done this is because we re-render the button when it’s clicked so you can see that it’s selected. I tried messing around with it with a solution following what you outlined but it seems that selected functionality is lost. 

why the timeout? seems like some workaround? add a comment about it.
-	Timeout is set so that the page renders before the word is spoken, allowing for the user to gain focus on the page. Added comment for useeffect.

There seems to be A LOT of duplication between this and BottomInput.
Why do we need this?
Let's find a way of reusing BottomInput instead of doing all the duplication?
-	You are absolutely correct! Now BottomInput.js is just used in AudioExerciseOne.js and BotInput.js has been removed. 

the name of displayedAudioExperimentPopup is not intention revealing. what is it standing for? is it expressing the fact that the audioExtensionPopupHasAlreadyBeenDisplayed?
-	Yes that’s what its standing for. We just followed the naming convention from the extension keys in localstorage.